libraries/outputs.rb

```diff
@@ -24,19 +24,42 @@ def self.configure_outputs(node) # rubocop:disable Metrics/PerceivedComplexity,
         else
           [CernerSplunk.my_cluster(node)]
         end.each do |(cluster, bag)|
+          if bag['indexer_discovery'] == true
+            Chef::Log.warn "Configured ['receivers'] in cluster #{cluster} will be ignored since ['indexer_discovery'] is set to true." if bag['receivers']
+
+            indexer_discovery_settings = ((bag['indexer_discovery_settings'] && bag['indexer_discovery_settings']['outputs_configs']) || {}).reject do |k, _|
+              k.start_with?('_cerner_splunk')
+            end
+            output_stanzas["indexer_discovery:#{cluster}"] = indexer_discovery_settings
+
+            fail "master_uri is missing in the cluster databag: #{cluster}" if bag['master_uri'].nil? || bag['master_uri'].empty?
+
+            output_stanzas["indexer_discovery:#{cluster}"]['master_uri'] = bag['master_uri']
+            encrypt_password = CernerSplunk::ConfTemplate::Transform.splunk_encrypt node: node
+
+            pass =
+              if bag['indexer_discovery_settings'] && bag['indexer_discovery_settings']['pass4SymmKey']
+                bag['indexer_discovery_settings']['pass4SymmKey']
+              else
+                'changeme'
+              end
+            output_stanzas["indexer_discovery:#{cluster}"]['pass4SymmKey'] = CernerSplunk::ConfTemplate.compose encrypt_password, CernerSplunk::ConfTemplate::Value.constant(value: pass)
+            output_stanzas["tcpout:#{cluster}"] = bag['tcpout_settings'] || {}
+            output_stanzas["tcpout:#{cluster}"]['indexerDiscovery'] = cluster
+            next
+          end
+
           port = bag['receiver_settings']
           port = port['splunktcp'] if port
           port = port['port'] if port
           receivers = bag['receivers']
 
-          if !receivers || receivers.empty? || !port
-            Chef::Log.warn "Receiver settings missing or incomplete in configured cluster data bag: #{cluster}"
-          else
-            output_stanzas["tcpout:#{cluster}"] = {}
-            output_stanzas["tcpout:#{cluster}"]['server'] = receivers.collect do |x|
-              x.include?(':') ? x : "#{x}:#{port}"
-            end.join(',')
-          end
+          fail "Receiver settings missing or incomplete in configured cluster data bag: #{cluster}" if !receivers || receivers.empty? || !port
```

Comment:
Being a more non-passive change, let's keep the warning for the receivers being missing for now. I'm picturing a use case, where I as an owner of forwarders define a "cluster" as my cluster to define sets of apps that should be on my servers, but use additional "clusters" to define where the logs should go.

But let's do keep the failure for if I've configured indexer discovery and do not specify the master URI.

Reply:
Updated here: 241d83680bfa98a7da6074c0b51ec25631f63135

I will squash once I get all the +1s.